### PR TITLE
Crash creating automatic backup of Custom Node with special Characters

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR is to address the crash issues that were occurring due to special or non-printable characters in the Custom Node name. 
https://jira.autodesk.com/browse/QNTM-3928.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner @QilongTang @Racel @alfarok 

### FYIs

@ColinDayOrg

Error window for no category:
categoryerror

Error window for invalid file name:
invalidname

### CODE REVIEW

File: src/DynamoCoreWpf/UI/Prompts/FunctionNamePrompt.xaml.cs [hunk 1]
```diff
@@ -49,6 +57,18 @@ void OK_Click(object sender, RoutedEventArgs e)
           }
       }
 
+      // Check if the file name contains any special non-printable chatacters.
+      bool IsFileNameValid(string fileName)
+      {
+          // Some other extra characters that are to be checked. 
+          char[] invalidCharactersFileName = { '#', '%', '&', '.', ' ' };
```
Comment (reddyashish) [inline src/DynamoCoreWpf/UI/Prompts/FunctionNamePrompt.xaml.cs]: Added these extra characters that were not being checked by the Path.GetInvalidFileNameChars() function. 
Made sure these special characters are not allowed: https://docs.microsoft.com/en-us/previous-versions/6f69bkd1(v=vs.140)

Comment (reddyashish) [reply src/DynamoCoreWpf/UI/Prompts/FunctionNamePrompt.xaml.cs]: Yes, moved the function to PathHelper.cs.

File: src/DynamoUtilities/PathHelper.cs [hunk 2]
```diff
@@ -146,5 +147,17 @@ public static bool isValidJson(string path, out string fileContents, out Excepti
...
             }
         }
+
+        // Check if the file name contains any special non-printable chatacters.
+        public static bool IsFileNameInValid(string fileName)
+        {
+            // Some other extra characters that are to be checked. 
+            char[] invalidCharactersFileName = { '#', '%', '&', '.', ' ' };
+
+            if (fileName.Any(f => Path.GetInvalidFileNameChars().Contains(f)) || fileName.IndexOfAny(invalidCharactersFileName) > -1)
```
Comment (alfarok) [inline src/DynamoUtilities/PathHelper.cs]: @reddyashish Can you add a comment linking to MSDN or something similar showing which characters are checked in `GetInvalidFileNameChars` vs the ones you have manually added?  I think you may have posted it in an earlier comment but don't see it anymore.

**Comment by Racel** [comment]
@reddyashish UI looks good. Please add tests.

File: src/DynamoUtilities/PathHelper.cs (hunk 2 above)
Comment (reddyashish) [reply src/DynamoUtilities/PathHelper.cs]: This Microsoft doc just shows some of the characters that are not valid for a file name. 
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getinvalidpathchars?view=netframework-4.7.2

I am also attaching a screenshot of all the characters in the array returned by GetInvalidFileNameChars() function. 

screen shot 2018-12-07 at 1 28 08 pm

The extra characters that were checked manually are { '#', '%', '&', '.', ' ' }.

**Comment by reddyashish** [comment]
Added the tests.

**Comment by QilongTang** [comment]
@reddyashish Can you do one favor for me and check if there is any file just for testing utilities? Because the place you put the test since does not seem to be the best, the tests there all tests serialization function which I think the test you end up using does not involve serialization

**Comment by QilongTang** [comment]
@reddyashish Only that last comment, otherwise, LGTM

**Comment by reddyashish** [comment]
@QilongTang Yes I will check that and move it.

**Comment by QilongTang** [comment]
@reddyashish NodeViewCustomization does not seem proper either, maybe `Dynamo\test\DynamoCoreTests\UtilityTests.cs`?

**Pull request merged**